File: palindromorecursivo.c
#include <stdio.h>
#include <stdbool.h>
#include <time.h>
#include <stdlib.h>
#include <string.h>

bool palindromo(char *s, int start, int end)
{
    if(start >= end)
    {
        return true;
    } // end if 

    if(s[start] != s[end])
    {
        return false;
    } // end if 

    return palindromo(s, start + 1, end -1);
} //end palindromo 

int main(void)
{
    // definir dados
    char s[80];

    printf("Digite uma string: ");
    scanf("%s", s);

    // funcoes auxiliares
    if(palindromo(s, 0, strlen(s) - 1))
    {
        printf("1\n");
    }
    else
    {
        printf("0\n");
    } // end if 
} // end main( )

File: Java/palindromorecursivo.java
import java.util.Scanner;
import java.util.Locale;

public class Main
{
    public static boolean palindromo(String s, int start, int end)
    {
        if(start >= end)
        {
            return true;
        } // end if

        if(s.charAt(start) != s.charAt(end))
        {
            return false;
        } // end if 

        return palindromo(s, start + 1, end - 1);
    } // end palindromo( )

    public static void main(String[] args)
    {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        // definir dados
        String s;
        System.out.print("Digite uma string: ");
        s = sc.nextLine();

        // funcoes auxiliares
        if(palindromo(s, 0, s.length() - 1))
        {
            System.out.println("1");
        }
        else
        {
            System.out.println("0");
        } // end if
        sc.close();
    } // end main( )
} // end class( )